### PR TITLE
Brendan lynn changes

### PR DESCRIPTION
**Pull request by currycoder**
- [X] Is this Pull Request ready for review?
- [X] Do the tests pass?
- [X] Have the docs been updated?

**What does this Pull Request do?**
A few changes necessary to build lynn:
- Added an extensions to django's test framework to make selenium acceptance testing easier.
- Break out autocomplete requirements in to a setuptools extra, such that it must now be explicitly installed with `pip install gn-django[autocomplete]`
- Document developing with django debug toolbar.
- Make default gulp file more resilient to less complilation errors.
- Add `ImproperlyConfigured` and `NotImplementedError` exceptions.
- Add some more general utility functions.
- Upgrade to django 1.11.5 - for minor feature additions.  There should be no backward compatibility breaking changes.

**Where can the reviewer see this in action?**
On lynn http://chloe.bn.gntech.systems:8080/

**How should this be tested?**
Existing tests are still all passing.

### CODE REVIEW

File: gn_django/url/utils.py [hunk 1]
```diff
@@ -60,3 +60,20 @@ def convert_camelcase_to_slugified(camelcase):
     slugified = re.sub('([a-z0-9])([A-Z])', r'\1-\2', slugified).lower()
     return slugified
 
+def convert_to_camelcase(to_convert):
```
Comment (mrmonkington) [inline gn_django/url/utils.py]: what about 'Kennedy, Mark'?

Comment (mrmonkington) [reply gn_django/url/utils.py]: And what about 'Hello       There'?

Basically use a regexp:

```python
re.replace('[^A-Za-z0-9]+', ' ').title(),replace(' ', '')
```

(assuming you want to preserve numbers)

File: docs/source/django_projects/static_files.rst [hunk 2]
```diff
@@ -47,12 +47,16 @@ The ``gulpfile.js`` that comes with the binary looks like this (excluding the in
 at the top)::
 
   gulp.task('compile', function () {
+    l.on('error',function(e){
```
[the remark this answers is not in the document]
Comment (currycoder) [reply docs/source/django_projects/static_files.rst]: Sorted

File: docs/source/packages/test.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+====
+
+The ``test`` package contains functionality for aiding with testing python/django
+libraries and applications.
+
+Acceptance tests in django
+--------------------------
+
+Acceptance tests ensure that some functionality is working and usable for an 
+end user.  In our case, this is best achieved by using automated browser testing
```
[the remark this answers is not in the document]
Comment (currycoder) [reply docs/source/packages/test.rst]: Never gonna happen.

File: gn_django/url/utils.py (hunk 1 above)
Comment (currycoder) [reply gn_django/url/utils.py]: Good point, dunnit.

**Pull request merged**